consensus/packets/tristatebitset.go

```diff
@@ -60,13 +63,151 @@ func (dbs *TriStateBitSet) ApplyChanges(changes []BitSetCell) {
 }
 
 func (dbs *TriStateBitSet) Serialize() ([]byte, error) {
-	return nil, nil
+	var firstByte uint8
+	var secondByte uint8
+	if dbs.CompressedSet {
+		firstByte = 0x01
+	} else {
+		firstByte = 0x00
+	}
+
+	array, err := dbs.cellsToBitArray()
+	if err != nil {
+		return nil, errors.Wrap(err, "[ Serialize ] failed to get bitarray from cells")
+	}
+
+	tmpLen := array.Len()
+	totalSize := int(math.Round(float64((tmpLen*2)/8))+0.5) + 1 // first byte
+	var result *bytes.Buffer
+	firstByte = firstByte << 1
+	if bits.Len(tmpLen) > 6 {
```

Comment:
Move each branch to separate func, e.g. `serializeHigh` and `serializeLow`. Plus, move magic number `6` to constants

Reply:
done